pkg/kubelet/kubelet.go

```diff
@@ -1975,7 +1975,7 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 		return false, nil
 	}
 
-	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && isPodResizeInProgress(pod, &apiPodStatus) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && isPodResizeInProgress(pod, podStatus) {
```

Comment:
Question about the comment below: why do we need to periodically call PLEG to update the cache. I'm not familiar with the original implementation of pod resizing, so likely I missed the point here.

Reply:
Thanks for adding the comment I had missed (important in hingsight): https://github.com/kubernetes/kubernetes/pull/102884#discussion_r817178710 . Eventually I hope EventedPLEG will take care of this and replace the hack.

Reply:
https://github.com/kubernetes/kubernetes/pull/123941 is attempting to fix this, but I'm skeptical that that approach is feasible. I think we have 2 options:
1. Augment the CRI API, either to include resources in ListContainers, or add something like ListContainerStatuses
2. Add some way to indicate that a container is being "watched", and should be reinspected.

The current approach is basically a more manual approach to 2, but I'm thinking about a way to better integrate it into the relist loop.